### PR TITLE
MM-53115 Fix ridiculous Babel bug by removing unused code 

### PR DESCRIPTION
**Pull request by hmhealey**
#### Summary
There's a more thorough explanation of this bug starting from [this post](https://community.mattermost.com/core/pl/my19ijhsciyk5fmr5xnhke937r) in that thread which goes into more detail about the issue, but basically, both `UpgradeModal` and `UpgradeBanner` have the following code
```js
const analytics = useSelector(getAdminAnalytics);
const serverTotalUsers = analytics?.TOTAL_USERS || 0;
```
which they never use the result of. For whatever reason, the updated version of Babel from https://github.com/mattermost/mattermost/pull/23543 rewrites that to call the selector, discards the result of it, and then call `analytics?.TOTAL_USERS` without having ever declared `analytics` which causes an error. After figuring that all out, thankfully, it was easy enough to fix by just removing the code that wasn't actually used.

I also decided to turn on the TS version of `no-unused-vars` in Playbooks to ensure we aren't running into this sort of thing anywhere else.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-53115

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by crspeller — APPROVED** [review]
Wow that was quite the bug. Good hunting!

File: webapp/playbooks/src/components/actions_modal_action_children.tsx [hunk 1]
```diff
@@ -49,7 +47,6 @@ interface OptionType {
 
 export const RunPlaybookChildren = ({playbookId, onUpdate, editable}: RunPlaybookProps) => {
     const {formatMessage} = useIntl();
-    const portalEl = useFloatingPortalNode();
```
Comment (lieut-data) [inline webapp/playbooks/src/components/actions_modal_action_children.tsx:52]: Is there by chance a weird side effect of calling `useFloatingPortalNode` that this code relies upon? (cc @calebroseland)

**Review by lieut-data — APPROVED** [review]

File: webapp/playbooks/src/components/actions_modal_action_children.tsx (hunk 1 above)
Comment (calebroseland) [reply webapp/playbooks/src/components/actions_modal_action_children.tsx:52]: Should be safe to remove. Shouldn't have any dependencies on useFloatingPortalNode side effects. It explicitly depends on our own modal infra (`#root-poral`—this is static in the app and always there).

**Review by calebroseland — APPROVED** [review]

**Pull request merged**